plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java

```diff
@@ -295,9 +330,17 @@ public Map<String, AzureStorageSettings> refreshAndClearCache(Map<String, AzureS
     }
 
     @Override
-    public void close() {
+    public void close() throws IOException {
         this.clients.values().forEach(this::closeInternally);
         this.clients.clear();
+        this.executor.shutdown();
+        try {
+            if (this.executor.awaitTermination(30, TimeUnit.SECONDS) == false) {
+                logger.warning("The executor was not shutdown gracefuly with 30 seconds");
```

Comment:
Maybe worth doing a `shutdownNow()` that will interrupt any threads that haven't stopped. Also fixed a typo.

```suggestion
                this.executor.shutdownNow();
                logger.warning("The executor was not shutdown gracefully with 30 seconds");
```

Reply:
fixed